pkgs/build-support/buildenv/default.nix

```diff
@@ -34,12 +34,13 @@
 }:
 
 runCommand name
-  { inherit manifest ignoreCollisions passthru pathsToLink extraPrefix postBuild buildInputs;
-    pkgs = builtins.toJSON (map (drv: {
-      paths = [ drv ]; # FIXME: handle multiple outputs
-      priority = drv.meta.priority or 5;
-    }) paths);
-    preferLocalBuild = true;
+  rec { inherit manifest ignoreCollisions passthru pathsToLink extraPrefix postBuild buildInputs;
+        pkgs = builtins.toJSON (map (drv: {
+          paths = [ drv ]; # FIXME: handle multiple outputs
+          priority = drv.meta.priority or 5;
+        }) paths);
+        preferLocalBuild = true;
+        passAsFile = if builtins.stringLength pkgs >= 128*1024 then [ "pkgs" ] else [];
```

Comment:
Hmm, in future, I would like for stdenv to filter `[]` and `""` values as well as `null`s. It will help with `lib.optional*` functions, especially for platform-specific conditions.